qiskit_experiments/calibration_management/update_library.py

```diff
@@ -112,32 +111,27 @@ def update(
             parameter: The name of the parameter in the calibrations to update.
             schedule: The ScheduleBlock instance or the name of the instance to which the parameter
                 is attached.
-            result_index: The result index to use. By default search entry by name.
+            result_index: The result index to use which defaults to -1.
             group: The calibrations group to update. Defaults to "default."
 
         Raises:
             CalibrationError: If the analysis result does not contain a frequency variable.
         """
-        if result_index is None:
-            result = [
-                r for r in exp_data.analysis_results() if r.name.startswith(PARAMS_ENTRY_PREFIX)
-            ][0]
-        else:
-            result = exp_data.analysis_results(index=result_index)
-
-        if cls.__fit_parameter__ not in result.extra["popt_keys"]:
-            raise CalibrationError(
-                f"{cls.__name__} updates from analysis classes "
-                f"which report {cls.__fit_parameter__} in popt."
-            )
-
-        param = parameter
-        value = result.value.value[result.extra["popt_keys"].index(cls.__fit_parameter__)]
+        value = BaseUpdater.get_value(exp_data, cls.__fit_parameter__, result_index)
```

Comment:
perhaps we need an option to get fit parameter name from user input rather than class member?

Reply:
I think for now we are okay. Perhaps in a follow up PR. Subclasses override `update` anyways.